src/compiler/checker.ts

```diff
@@ -24607,6 +24621,10 @@ namespace ts {
                     return getTypeFacts(leftType) & TypeFacts.Falsy ?
                         getUnionType([removeDefinitelyFalsyTypes(leftType), rightType], UnionReduction.Subtype) :
                         leftType;
+                case SyntaxKind.QuestionQuestionToken:
+                    return getTypeFacts(leftType) & TypeFacts.EQUndefinedOrNull ?
+                        getUnionType([getNonNullableType(leftType), rightType], UnionReduction.Subtype) :

```

Comment:
We are getting type here without `null` or `undefined` but when we downlevel we only check for `null` is this intended. cc: @DanielRosenwasser

Reply:
`== null` checks for both `null` and `undefined`. So `x == null` is almost equivalent to `x === null || x === undefined`.

The only gotcha is that `document.all == null` is true but `document.all === null || document.all === undefined` is false.

Reply:
Somehow I missed `==` and thought of that as `===`